docs/react/guides/deploy-nextjs-to-vercel.md

```diff
@@ -32,4 +32,45 @@ When everything is updated appropriately, for our `tuskdesk` example we would se
 
 ![image](./next-deploy-vercel-2.png)
 
+## Skipping build if the application is not affected
+
+One of the core features of Nx is to run code quality checks and builds only for projects that are affected by recent code changes. We can use [Vercel's ignored build step feature](https://vercel.com/docs/platform/projects#ignored-build-step) to only build our application if it is affected.
+
+We are going to achieve this by creating a shell script in our Nx workspace that will be invoked by Vercel.
+
+```sh
+
+# Name of the app to check. Change this to your application name!
+APP=tuskdesk
+# Determine version of Nx installed
+NX_VERSION=$(node -e "console.log(require('./package.json').devDependencies['@nrwl/workspace'])")
+
+# Install @nrwl/workspace in order to run the affected command
+yarn add -D @nrwl/workspace@$NX_VERSION --prefer-offline
+
+# Run the affected command, comparing latest commit to the one before that
+npx nx affected:apps --plain --base HEAD~1 --head HEAD | grep $APP -q
+
+# Store result of the previous command (grep)
+IS_AFFECTED=$?
+
+if [ $IS_AFFECTED -eq 1 ]; then
+  echo "🛑 - Build cancelled"
+  exit 0
+elif [ $IS_AFFECTED -eq 0 ]; then
+  echo "✅ - Build can proceed"
+  exit 1
+fi
+```
+
+There are a few points worth noting about this script.
+
+Firstly, you might have noticed that we are running `yarn add` in this script, as a result running it will take some time. Having said that, it is usually still much faster than installing all dependencies and running the build of your application unconditionally.
+
+Secondly, this script only compares changes introduced in a single latest commit. So if you push multiple new commits - the script will only check if your application is affected by changes in the latest commit.
```

Comment:
We can mention that in the future there should be an environment variable in Vercel to determine what the`base` should be.